internal/interpreter/dynamic_component.rs

```diff
@@ -1592,23 +1676,21 @@ extern "C" fn subtree_index(component: ComponentRefPin) -> usize {
 unsafe extern "C" fn parent_node(component: ComponentRefPin, result: &mut ItemWeak) {
     generativity::make_guard!(guard);
     let instance_ref = InstanceRef::from_pin_ref(component, guard);
-    let parent_item_index = instance_ref
-        .component_type
-        .original
-        .parent_element
-        .upgrade()
-        .and_then(|e| e.borrow().item_index.get().cloned());
-    if let (Some(parent_offset), Some(parent_index)) =
-        (instance_ref.component_type.parent_component_offset, parent_item_index)
-    {
-        if let Some(parent) = parent_offset.apply(instance_ref.as_ref()) {
-            generativity::make_guard!(new_guard);
-            let parent_instance = InstanceRef::from_pin_ref(*parent, new_guard);
-            let parent_rc =
-                parent_instance.self_weak().get().unwrap().clone().into_dyn().upgrade().unwrap();
-            *result = ItemRc::new(parent_rc, parent_index).downgrade();
-        };
-    }
+
+    if let Some(ComponentKind::SubComponent { parent, item_tree_index }) = &instance_ref.parent() {
```

Comment:
In this function, instead of having the parent node stored in the component and use the ComponentVTable for it, you could just find out that you are requesting the parent of a `Embed` and then when getting the parent of the root of a embed, just return the node of the `Embed`.

That way, you don't need to add a set_parent_node to the ComponentVTable and the parent in each component.

Reply:
The ComponentVTable return none for the parent if there is no parent.
But when you query it parent, you know you are querying the parent of a given embedded component, so if it returns none, you can just return the node for the embed.

Reply:
yeah... i see.